chromadb/test/ef/test_nomic_ef.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+import pytest
+
+# import requests
+# from requests import HTTPError
+# from requests.exceptions import ConnectionError
+
+from chromadb.utils.embedding_functions import NomicEmbeddingFunction
+
+
+def test_nomic() -> None:
```

Comment:
Let's add couple of negative test cases:

- Empty API key
- Error response (mock server)
- Too many texts


For the mock server:

```bash
pip install pytest_httpserver>=1.0.10
```

```py
import pytest
from pytest_httpserver import HTTPServer
import json
from unittest.mock import patch

with HTTPServer() as httpserver:
    # Define the response
    httpserver.expect_oneshot_request("/embeddings/text",method="POST").respond_with_data(
        json.dumps({"error": "some error message"}), # adjust to fit Nomic API response
        status=400,
        
    )
    with patch.object(NomicEmbeddingFunction, '_api_url', f"http://{httpserver.host}:{httpserver.port}/embeddings/text",):
        nomic_instance = NomicEmbeddingFunction()
        with pytest.raises(Exception):
            nomic_instance(["test text"])
```

Reply:
Added these tests as well as one for missing model name, but did not include the "too many texts" test. Nomic responded saying: 

- "You can send as many as you'd like per request, they will get processed in parallel. It's recommended you break it up yourself into several requests or use the Nomic python client because you will see network latency due to the large request/response size if you send dozens of megabytes of text in a single request" 

So I am not sure 1) what we want consider too large 2) best way to check for that 3) if you still want this

Reply:
Also -- should I add pytest_httpserver to the projects requirements file and check it in as part of this PR?

Reply:
Yes, please add it to the `requirements_dev`.

We do have a limitation in Chroma API regarding the maximum number of embeddings, but we generally don't tie that to the embeddings functions batch size. So you can leave it as is for now.